app/src/main/java/com/naman14/timberx/ui/fragments/MainFragment.kt

```diff
@@ -97,42 +95,57 @@ class MainFragment : Fragment() {
         when (item.itemId) {
             R.id.menu_item_about -> {
-                AboutDialog().show(activity!!.supportFragmentManager, "AboutDialog")
+                val context = activity ?: return false
+                AboutDialog().show(context.supportFragmentManager, "AboutDialog")
             }
         }
         return super.onOptionsItemSelected(item)
     }
 
     private fun setupViewPager(viewPager: ViewPager) {
-        val adapter = Adapter(childFragmentManager)
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_SONGS.toString(), null)), this.getString(R.string.songs))
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_ALBUMS.toString(), null)), this.getString(R.string.albums))
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_PLAYLISTS.toString(), null)), this.getString(R.string.playlists))
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_ARTISTS.toString(), null)), this.getString(R.string.artists))
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_FOLDERS.toString(), null)), this.getString(R.string.folders))
-        adapter.addFragment(MediaItemFragment.newInstance(MediaID(TimberMusicService.TYPE_ALL_GENRES.toString(), null)), this.getString(R.string.genres))
+        val res = context?.resources ?: return
+        val adapter = Adapter(childFragmentManager).apply {
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_SONGS.toString(), null)),
+                    res.getString(R.string.songs)
+            )
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_ALBUMS.toString(), null)),
+                    res.getString(R.string.albums)
+            )
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_PLAYLISTS.toString(), null)),
+                    res.getString(R.string.playlists)
+            )
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_ARTISTS.toString(), null)),
+                    res.getString(R.string.artists)
+            )
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_FOLDERS.toString(), null)),
+                    res.getString(R.string.folders)
+            )
+            addFragment(
+                    MediaItemFragment.newInstance(MediaID(TYPE_ALL_GENRES.toString(), null)),
+                    res.getString(R.string.genres)
+            )
+        }
         viewPager.adapter = adapter
     }
 
     internal class Adapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
-        private val mFragments = ArrayList<Fragment>()
-        private val mFragmentTitles = ArrayList<String>()
+        private val fragments = ArrayList<Fragment>()
```

Comment:
This is actually the bug causing https://github.com/naman14/TimberX/issues/11

Reply:
@Zhuinden a bug should be fixed in a separate PR

Reply:
Ya I'm just making a note, nevermind me in this regard :smile:

Reply:
Yeah, I will address this